src/colorizer/Dataset.ts

```diff
@@ -151,8 +170,43 @@ export default class Dataset {
     }
   }
 
+  /**
+   * Gets the feature's units if it exists; otherwise returns an empty string.
+   */
   public getFeatureUnits(name: string): string {
-    return this.features[name].units;
+    return this.features.get(name)?.units || "";
+  }
+
+  /**
+   * Returns the FeatureType of the given feature, if it exists.
+   * @param name Feature name to retrieve
+   * @throws An error if the feature does not exist.
+   * @returns The FeatureType of the given feature (categorical, continuous, or discrete)
+   */
+  public getFeatureType(name: string): FeatureType {
+    const featureData = this.features.get(name);
+    if (!featureData) {
+      throw new Error(`Feature ${name} does not exist.`);
```

Comment:
I'm a little confused why you have this error check here but not in `getFeatureData`. it seems like `getFeatureData` should be checking for wether the feature exists and any function that is trying to unpack that data should call `getFeatureData` instead of `this.features.get(name)`